library/src/scala/annotation/internal/WitnessNames.scala

```diff
@@ -0,0 +1,54 @@
+package scala.annotation
+package internal
+
+/** An annotation that is used for marking type definitions that should get
+ *  context bound companions. The scheme is as follows:
+ *
+ *  1. When desugaring a context-bounded type A, add a @WitnessNames(n_1, ... , n_k)
+ *  annotation to the type declaration node, where n_1, ..., n_k are the names of
+ *  all the witnesses generated for the context bounds of A. This annotation will
+ *  be pickled as usual.
+ *
+ *  2. During Namer or Unpickling, when encountering a type declaration A with
+ *  a WitnessNames(n_1, ... , n_k) annotation, create a CB companion `val A` with
+ *  rtype `<context-bound-companion>`[ref_1 | ... | ref_k] where ref_i is a TermRef
+ *  with the same prefix as A and name n_i. Except, don't do this if the type in
+ *  question is a type parameter and there is already a term parameter with name A
+ *  defined for the same method.
+ *
+ *  ContextBoundCompanion is defined as an internal abstract type like this:
+ *
+ *     type `<context-bound-companion>`[-Refs]
+ *
+ *  The context bound companion's variance is negative, so that unons in the
+ *  arguments are joined when encountering multiple definfitions and forming a glb.
```

Comment:
What's a glb?

Reply:
greatest lower bound type

Reply:
Greatest lower bound (it's established in the dotty context)